pandas/core/internals.py

```diff
@@ -1929,6 +1903,9 @@ def should_store(self, value):
 
 
 class DatetimeLikeBlockMixin(object):
+    # We add a dummy `freq` attribute here to make these block subclasses
+    # behave more like the DatetimelikeIndexOpsMixin implementations
+    freq = None
```

Comment:
I don't understand the purpose for this. Why should blocks behave more like DatetimelineIndexOpsMixin ?

Reply:
I see the point for adding `tz` (for a little bit of deduplication), but `freq` is never used? Why adding it?

Reply:
>I don't understand the purpose for this. Why should blocks behave more like DatetimelineIndexOpsMixin ?

@jorisvandenbossche Two parallel strands here.  

a) #19147 and a bunch of others are moving towards having Series ops delegate to the Index subclass methods, which have careful implementations of timezone and type checking.  But longer-term, it would be better for these methods to just delegate to e.g. `self._data.blocks[0].__add__` (probably accessed more prettily) and have the implementations in the blocks.  This will also make it easier for DataFrame ops to delegate correctly, see #18824.  So ideally e.g. TimedeltaBlock and TimedeltaIndex would both inherit their arithmetic/comparison methods from TimedeltaArray or TimedeltaVector or something.

b) #19174 has PeriodArray subclass DatetimelikeOpsMixin, notes that it would be preferable to have the mixin class refactor out index-specific ops and separate the more array-like ops.  I've been working along those lines, have found that small namespaces differences are the main thing standing in the way of this refactoring.  So partly this is trying to smooth out those differences in a piece-meal way.

That definitely answers _some_ question.  Does it answer the right question?

 > I see the point for adding tz (for a little bit of deduplication), but freq is never used? Why adding it?

Purely for compat.  This way `DatetimeBlock._box_func == DatetimeTZBlock._box_func == DatetimeIndex._box_func`.

Reply:
Yes, I understand that we are thinking about that direction, but those things would be added in an array class, not the block. Anyhow, let's change this if we actually get there, not prematurely, as it is now only adding unused code (I mean the `freq` here).

Reply:
Sounds good.